test/unit/org/apache/cassandra/metrics/ViewWriteMetricsTest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.metrics;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.io.IOException;
+import java.util.function.LongSupplier;
+
+import org.apache.cassandra.SchemaLoader;
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.exceptions.ConfigurationException;
+import org.apache.cassandra.schema.Schema;
+import org.apache.cassandra.service.EmbeddedCassandraService;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.Session;
+
+public class ViewWriteMetricsTest extends SchemaLoader
+{
+    private static Session session;
+
+    @BeforeClass()
+    public static void setup() throws ConfigurationException, IOException
+    {
+        Schema.instance.clear();
+
+        EmbeddedCassandraService cassandra = new EmbeddedCassandraService();
+        cassandra.start();
+
+        Cluster cluster = Cluster.builder().addContactPoint("127.0.0.1").withPort(DatabaseDescriptor.getNativeTransportPort()).build();
+        session = cluster.connect();
+    }
+
+    @Test
+    public void testMetricsCleanupOnDrop()
+    {
+        String keyspace = "junit";
+        String table = "viewwritemetrics";
+        String view = "viewwritemetrics_materialized_view";
+
+        CassandraMetricsRegistry registry = CassandraMetricsRegistry.Metrics;
+        LongSupplier count = () -> registry.getNames().stream().filter(n -> n.contains(view)).count();
+
+        // no metrics before create
+        assertEquals(0, count.getAsLong());
+
+        // TODO the ViewWriteMetrics metrics are never created
```

Comment:
@stillalex There are two kinds of metrics for MVs. The first represents the conditionally registered metrics in `TableMetrics`, for example `viewReadTime`. Then you have `ViewWriteMetrics` which is created on the `StorageProxy` and spans tables. (I think I mentioned the latter a bit too carelessly in one of my earlier comments on the Jira. Apologies.)

In any case, all I think we need to do is drop this test at the end of `TableMetricsTest` and we're golden. That will ensure that even MVs-specific metrics in `TableMetrics` are removed. Does that make sense?

Reply:
I'll move the test over, sure. the trouble is I'm trying to verify (manually) that those metrics are actually created, otherwise that part of the code is not covered, so the test is not actually testing anything :)

Reply:
An MV actually has fewer metrics than a normal table, due to `ViewLockAcquireTime` and `ViewReadTime` being present only for the latter. This test is still valuable, as it makes sure removal still works when those aren't present (i.e. the conditional logic in `releaseMetric()`).

Either way, I don't think we need to worry about `ViewWriteMetrics`.

Reply:
If you want to verify here that `ViewLockAcquireTime` and `ViewReadTime` are created only the base table and not an MV, you could filter the names on each of them, etc.